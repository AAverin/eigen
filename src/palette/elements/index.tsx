export * from "./Cards"
export * from "./Touchable"
export * from "./Typography"
