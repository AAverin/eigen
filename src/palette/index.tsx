export * from "./atoms"
export * from "./organisms"
export * from "./elements"
export * from "./Theme"
