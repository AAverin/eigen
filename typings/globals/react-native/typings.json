{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts",
    "raw": "registry:dt/react-native#0.29.0+20160719232534",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts"
  }
}
