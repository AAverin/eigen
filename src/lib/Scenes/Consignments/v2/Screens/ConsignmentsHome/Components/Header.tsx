import { Box, Button, Sans, Spacer } from "@artsy/palette"
import React from "react"

interface HeaderProps {
  onCTAPress?: () => void
}

export const Header: React.FC<HeaderProps> = ({ onCTAPress }) => {
  return (
    <Box px={2} mt={3}>
      <Box>
        <Sans size="8" textAlign="center" px={2}>
          Sell Art From Your Collection
        </Sans>

        <Spacer my={0.5} />

        <Sans size="4" textAlign="center">
          Reach art buyers all over the world.
        </Sans>
      </Box>

      <Spacer mb={2} />

      <Button variant="primaryBlack" block onPress={onCTAPress}>
        <Sans size="3" weight="medium">
          Start selling
        </Sans>
      </Button>
    </Box>
  )
}
