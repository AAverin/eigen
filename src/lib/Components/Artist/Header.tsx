import { Button } from "@artsy/palette"
import { HeaderFollowArtistMutation } from "__generated__/HeaderFollowArtistMutation.graphql"
import colors from "lib/data/colors"
import React from "react"
import { StyleSheet, TextStyle, View } from "react-native"
import { commitMutation, createFragmentContainer, graphql, RelayProp } from "react-relay"
import { Schema, Track, track as _track } from "../../utils/track"
import Headline from "../Text/Headline"
import SerifText from "../Text/Serif"

import { Header_artist } from "__generated__/Header_artist.graphql"

interface Props {
  artist: Header_artist
  relay: RelayProp
}

interface State {
  followersCount: number
  isFollowedChanging: boolean
}

const track: Track<Props, State> = _track

@track()
class Header extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      isFollowedChanging: false,
      followersCount: props.artist.counts.follows as number,
    }
  }

  render() {
    const artist = this.props.artist
    return (
      <View style={{ paddingTop: 20 }}>
        <Headline style={[styles.base, styles.headline]}>{artist.name}</Headline>
        {this.renderByline()}
        {this.renderFollowersCount()}
        {this.renderFollowButton()}
      </View>
    )
  }

  renderFollowButton() {
    if (this.props.artist.isFollowed !== null) {
      return (
        <Button
          variant={this.props.artist.isFollowed ? "secondaryOutline" : "primaryBlack"}
          block
          width={100}
          loading={this.state.isFollowedChanging}
          onPress={this.handleFollowChange.bind(this)}
        >
          {this.props.artist.isFollowed ? "Following" : "Follow"}
        </Button>
      )
    }
  }

  renderFollowersCount() {
    const count = this.state.followersCount
    const followerString = count + (count === 1 ? " Follower" : " Followers")
    return <SerifText style={[styles.base, styles.followCount]}>{followerString}</SerifText>
  }

  renderByline() {
    const artist = this.props.artist
    const bylineRequired = artist.nationality || artist.birthday
    if (bylineRequired) {
      return (
        <View>
          <SerifText style={styles.base}>{this.descriptiveString()}</SerifText>
        </View>
      )
    } else {
      return null
    }
  }

  descriptiveString() {
    const artist = this.props.artist
    const descriptiveString = (artist.nationality || "") + this.birthdayString()
    return descriptiveString
  }

  birthdayString() {
    const birthday = this.props.artist.birthday
    if (!birthday) {
      return ""
    }

    const leadingSubstring = this.props.artist.nationality ? ", b." : ""

    if (birthday.includes("born")) {
      return birthday.replace("born", leadingSubstring)
    } else if (birthday.includes("Est.") || birthday.includes("Founded")) {
      return " " + birthday
    }

    return leadingSubstring + " " + birthday
  }

  @track(props => ({
    action_name: props.artist.isFollowed ? Schema.ActionNames.ArtistUnfollow : Schema.ActionNames.ArtistFollow,
    action_type: Schema.ActionTypes.Tap,
    owner_id: props.artist.internalID,
    owner_slug: props.artist.slug,
    owner_type: Schema.OwnerEntityTypes.Artist,
  }))
  handleFollowChange() {
    const {
      relay,
      artist: { slug, id, isFollowed },
    } = this.props
    const { isFollowedChanging } = this.state

    if (isFollowedChanging) {
      return
    }

    this.setState(
      {
        isFollowedChanging: true,
      },
      () => {
        commitMutation<HeaderFollowArtistMutation>(relay.environment, {
          onCompleted: () => this.successfulFollowChange(),
          mutation: graphql`
            mutation HeaderFollowArtistMutation($input: FollowArtistInput!) {
              followArtist(input: $input) {
                artist {
                  id
                  isFollowed
                }
              }
            }
          `,
          variables: {
            input: {
              artistID: slug,
              unfollow: isFollowed,
            },
          },
          optimisticResponse: {
            followArtist: {
              artist: {
                id,
                isFollowed: !isFollowed,
              },
            },
          },
          onError: () => this.failedFollowChange(),
        })
      }
    )
  }

  @track(props => ({
    action_name: props.artist.isFollowed ? Schema.ActionNames.ArtistFollow : Schema.ActionNames.ArtistUnfollow,
    action_type: Schema.ActionTypes.Success,
    owner_id: props.artist.internalID,
    owner_slug: props.artist.slug,
    owner_type: Schema.OwnerEntityTypes.Artist,
  }))
  successfulFollowChange() {
    // callback for analytics purposes
    this.setState({
      isFollowedChanging: false,
    })
  }

  @track(props => ({
    action_name: props.artist.isFollowed ? Schema.ActionNames.ArtistFollow : Schema.ActionNames.ArtistUnfollow,
    action_type: Schema.ActionTypes.Fail,
    owner_id: props.artist.internalID,
    owner_slug: props.artist.slug,
    owner_type: Schema.OwnerEntityTypes.Artist,
  }))
  failedFollowChange() {
    // callback for analytics purposes
    this.setState({
      isFollowedChanging: false,
    })
  }
}

interface Styles {
  base: TextStyle
  headline: TextStyle
  followCount: TextStyle
}

const styles = StyleSheet.create<Styles>({
  base: {
    textAlign: "center",
  },
  headline: {
    fontSize: 14,
  },
  followCount: {
    color: colors["gray-semibold"],
    marginBottom: 30,
  },
})

export default createFragmentContainer(Header, {
  artist: graphql`
    fragment Header_artist on Artist {
      id
      internalID
      slug
      isFollowed
      name
      nationality
      birthday
      counts {
        follows
      }
    }
  `,
})
