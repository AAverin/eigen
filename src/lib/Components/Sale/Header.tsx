import * as React from "react"
import { Dimensions, NativeModules, StyleSheet, Text, TextStyle, View, ViewProperties, ViewStyle } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"
import OpaqueImageView from "../OpaqueImageView"
import Headline from "../Text/Headline"

class Header extends React.Component<RelayProps, {}> {
  constructor(props) {
    super(props)
    this.state = {}
  }

  render() {
    const { width, height } = Dimensions.get("window")
    const sale = this.props.sale
    return (
      <View>
        <View style={styles.header}>
          {this.props.showImage
            ? <OpaqueImageView
                // TODO: un-hardcode this (metaphysics isn't resolving sales' cover_image field property.)
                imageURL="https://d32dm0rphc51dk.cloudfront.net/3YueUJ2y1Vwx-YQJ82lG_w/wide.jpg"
                style={{ position: "absolute", top: 0, width, height }}
              />
            : null}
          <Headline style={styles.headline} numberOfLines={2}>
            {sale.name}
          </Headline>
        </View>
      </View>
    )
  }
}

export default createFragmentContainer(
  Header,
  graphql`
    fragment Header_sale on Sale {
      cover_image {
        href
      }
    }
  `
)

const isPad = Dimensions.get("window").width > 700

interface Styles {
  header: TextStyle
  headline: TextStyle
}

const styles = StyleSheet.create<Styles>({
  header: {
    marginTop: 15,
    height: 36,
    justifyContent: "center",
    alignItems: "center",
    paddingLeft: 40,
    paddingRight: 40,
  },
  headline: {
    textAlign: "center",
    fontSize: isPad ? 20 : 14,
  },
})

interface RelayProps {
  sale: {
    _id: string
    id: string
    name: string | null
  }
  showImage: boolean
}
