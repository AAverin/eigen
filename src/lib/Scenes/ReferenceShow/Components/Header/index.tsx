import React from "react"
import { View } from "react-native"

// TODO: implement

export class Header extends React.Component {
  render() {
    return <View />
  }
}
