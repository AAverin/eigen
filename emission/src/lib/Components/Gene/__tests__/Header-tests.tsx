import { NativeModules } from "react-native"

import React from "react"
import * as renderer from "react-test-renderer"

import Header from "../Header"

import { Theme } from "@artsy/palette"

beforeAll(() => {
  NativeModules.ARTemporaryAPIModule = { followStatusForGene: jest.fn() }
})

it("renders properly", () => {
  const gene = {
    id: "gene-deep-time",
    internalID: "gravity-id",
    gravityID: "deep-time",
    name: "Deep Time",
  }

  const header = renderer
    .create(
      <Theme>
        <Header gene={gene as any} shortForm={false} />
      </Theme>
    )
    .toJSON()
  expect(header).toMatchSnapshot()
})
