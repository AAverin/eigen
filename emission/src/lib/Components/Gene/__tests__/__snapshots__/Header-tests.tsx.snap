// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<Header
  gene={
    Object {
      "gravityID": "deep-time",
      "id": "gene-deep-time",
      "internalID": "gravity-id",
      "name": "Deep Time",
    }
  }
  shortForm={false}
/>
`;
